src/vivilux/visualize.py

```diff
@@ -43,6 +43,8 @@ def __init__(self, name: str,
         self.ax.set_ylim(0, self.ylim)
         self.ax.legend(range(numLines))
 
+        plt.ion() # Enable interactive plot for real-time updates
```

Comment:
Is this call appropriate? If the user wants to add the monitor but do some pre-training without turning it on first, then the user should turn on and off interactive mode simultaneously. Maybe the layer class should have a function that calls a "switch on/off" function in the monitor class for each of its instances.